ui/raidboss/data/03-hw/trial/thordan-ex.ts

```diff
@@ -0,0 +1,513 @@
+      alertText: (_data, _matches, output) => output.icePuddleYou!(),
+      outputStrings: {
+        icePuddleYou: {
+          en: 'Ice puddle on YOU',
+        },
+      },
+    },
+    {
+      id: 'ThordanEX Comet Puddles',
+      type: 'HeadMarker',
+      netRegex: { id: '000B' },
+      condition: Conditions.targetIsYou(),
+      alertText: (_data, _matches, output) => output.meteorYou!(),
+      outputStrings: {
+        meteorYou: {
+          en: '4x meteor puddles on YOU',
+        },
+      },
+    },
+    {
+      id: 'ThordanEX Fury Spear',
+      type: 'HeadMarker',
+      netRegex: { id: '0010' },
+      alarmText: (data, matches, output) => {
+        // Whoever is actively tanking Thordan must not stack,
+        // because they will be taking Heavenly Heel shortly after.
+        // If they stack, they will receive the physical vulnerability up and auto-die.
+        if (data.me !== data.mainTank)
+          return;
+        if (data.me === matches.target)
+          return output.spearYou!();
+        return output.spearMainTank!();
+      },
+      alertText: (data, matches, output) => {
+        if (data.me === data.mainTank)
+          return;
+        if (data.me === matches.target)
+          return output.spearYou!();
+        return output.spearOther!({ spearTarget: matches.target });
+      },
+      outputStrings: {
+        spearYou: {
+          en: 'Wild Charge on YOU',
+        },
+        spearMainTank: {
+          en: 'Wild Charge: STAY OUT',
+        },
+        spearOther: {
+          en: 'Wild Charge: Intercept ${spearTarget}',
+        },
+      },
+    },
+    {
+      id: 'ThordanEX Pure Of Soul',
+      type: 'StartsUsing',
+      netRegex: { id: '14B1', source: 'Ser Charibert', capture: false },
+      response: Responses.aoe(),
+    },
+    {
+      id: 'ThordanEX Single Spiral Thrust',
+      type: 'Ability',
+      netRegex: { id: '1018', source: 'Ser Vellguine' }, // Shared ability from all knights when they teleport in.
+      condition: (data) => data.phase === 5,
+      infoText: (_data, matches, output) => {
+        const knightNum = Directions.hdgTo8DirNum(parseFloat(matches.heading));
+        const knightDir = Directions.outputFrom8DirNum(knightNum);
+        const [dir1, dir2] = [knightDir, unsafeMap[knightDir]].sort();
+        if (dir1 === undefined || dir2 === undefined)
+          return;
+        return output.combined!({ dir1: output[dir1]!(), dir2: output[dir2]!() });
+      },
+      outputStrings: {
+        combined: {
+          en: '${dir1} / ${dir2} Unsafe',
+        },
+        ...Directions.outputStrings8Dir,
+      },
+    },
+    {
+      id: 'ThordanEX Faith Unmoving Off Center',
+      type: 'Ability',
+      netRegex: { id: '1018', source: 'Ser Grinnaux' }, // Shared ability from all knights when they teleport in.
+      condition: (data) => data.phase === 4,
+      delaySeconds: 6, // Grinnaux insta-casts Faith Unmoving 13s after appearing. Give ~7s of warning.
```

Comment:
Maybe give 6s of warning, so arms length etc can cover it as soon as the call happens? (if arms length works?)